### DISCUSSION TITLE
Ability to use "as" in new App Router

### DISCUSSION THREAD

**Discussion by mrjackyliang** [vercel/next.js#63115] [App Router]
When I upgraded one of my apps that is based on a slider, the new App Router completely broke things for me and the design I built around it. Using the pages router, I can store the state into the URL (more details below), and I want to hide the state away from the user.

```
await nextRouter.push(
  [
    ...baseUrl,
    `?searchHash=${searchHash}`,
    '&slideIndex=0',
  ].join(''),
  baseUrl.join(''),
);
```

Before we get to the idea of "bad practice" or "in favor of", first see what the use cases are here:
1. The search hash and slide index should not be cached as browser storage because the two fields are used to determine what information to show. In my instance, the user initializes a search. Then they would be routed to the detail page, where a slider would show all of the information, one by one, fully paged.
2. The URL changes based on which slide the user is on, in addition to the state being hidden, making the URL sharable since the state is hidden away from the user. They're able to simply copy and paste the URL and another user would be able to load the URL up easily.
3. The page would not refresh every time the URL was changed (a good thing). With the app router, the slider destroys itself and recreates it, causing an unnecessary burden on the client.

**Comment by kimfucious** [comment]
Yeah, I used to be able to do this, now I can't 😞 

```ts
                                const pathname = `/${path}/${item.author}/${item.slug}`;
                                router.push(
                                    {
                                        pathname: pathname,
                                        query: {
                                            hiddenInfo: "ugly stuff that shouldn't be seen",
                                        },
                                    },
                                    pathname
                                );
```

**Comment by khuezy** [comment]
Yes this would be great to have. My use case is for the modal interception. When a user loads an url that's nested in a modal, eg: `/photos/1`, then any navigation on that page causes the modal to be intercepted - which isn't expected behavior.
If navigation "as" exists, then you can navigate to a clone of the path, but via new route so the modal won't intercept.
